generator/internal/dart/dart.go

```diff
@@ -156,11 +156,22 @@ func resolveTypeName(message *api.Message, packageMapping map[string]string, imp
 	return messageName(message)
 }
 
+var messageRenames = map[string]string{
+	// 'Function' is a reserved keyword: https://dart.dev/language/keywords.
+	".google.cloud.functions.v2.Function": "CloudFunction",
```

Comment:
I expect clashes between field names and the Dart keywords.  We should think of a more generic way to avoid conflicts.  For example, could replace `Function` with `Function_` or `Function$`.  

- Something that starts with the original name seems like would work better for code completion
- It also raises fewer questions about "Is this a mangled name?": the weird characters are obvious mangling.
- And both `$` and `_` [1] should not appear in the proto so no conflicts if the same service defines both `Function` and `CloudFunction` ([1] the latter just by convention).

Reply:
I switched this over to using a (short) map of reserved words and resolving collisions by prefixing `$`. I don't really love how it looks but it is mechanical and, like you said, doesn't harm discovery.

In another PR, I may move the nested symbol concatenation to using `_` (currently `ReplicationStatus$UserManagedStatus$ReplicaStatus`). I looked into not using any concatenation, but there are too many short message names - `Policy` - and short, common enum names - `Type`, `State`. Perhaps messages could concatenate w/ `_` and enums could concatenatr w/o a separator? `Message_Message` vs `Message_MessageEnum`